app/src/main/java/org/simple/clinic/home/HomeScreenEffectHandler.kt

```diff
@@ -19,5 +23,16 @@ class HomeScreenEffectHandler @AssistedInject constructor(
   fun build(): ObservableTransformer<HomeScreenEffect, HomeScreenEvent> = RxMobius
       .subtypeEffectHandler<HomeScreenEffect, HomeScreenEvent>()
       .addAction(OpenFacilitySelection::class.java, uiActions::openFacilitySelection, schedulersProvider.ui())
+      .addTransformer(LoadCurrentFacility::class.java, loadCurrentFacility())
       .build()
+
+  private fun loadCurrentFacility(): ObservableTransformer<LoadCurrentFacility, HomeScreenEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .flatMap { userSession.requireLoggedInUser() }
+          .flatMap { facilityRepository.currentFacility(it) }
```

Comment:
In a recent PR (#1598), we added a way to inject a stream of the current facility as `Observable<Facility>`. In the final cleanup of this screen, we might want to switch to using this.